### ISSUE TITLE
Virtual memorycard saves mixed with instant saves

### ISSUE THREAD

**Issue by gui-don** [digitalLumberjack/recalbox-os#474]
I’ve had this problem for a long time without without getting to the bottom of it, until recently.

Sometimes, it look likes memorycard saves are erased for some reason. With Seiken Densetsu 3 (SNES), I succeed in duplicating the problem. Here are the steps:

- Playing a while ;
- Save using instant saves (save state) ;
- Go to a save point, save (into memorycard) ;

At this point, if you quit the game, everything is alright. Every saves are ok. But if:

- Load the previously save state ;
- Quit the game ;

Then memorycard save is erased (or one memorycard save is present, but it is always something from before the save state).

TL;DR: Loading a save state before exiting a game change the memorycard save to another one made before the save state. It seems memorycard saves and save states are linked.

On *Recalbox OS 3.3-beta7*

**Comment by digitalLumberjack** [comment]
Yes it is normal. When you make a save state, it changes the whole state of a game, the memory card also. Because look at this scenario : 
- Play
- save using save state
- go to a save point
- change state slot
- play
- save point

If the ingame saves would not be linked to the state, you would erase the memory of the first state you saved, it's not what you want.

**Comment by gui-don** [comment]
I see. It comes from a little misconception of save states for me. I didn’t know memory card were saved with everything else, and as you mention it, it makes sens.

So, I understand it’s wanted, because it can lead to unexpected behavior and safeguard losses. We certainly don’t want this. On the other hand, this very feature also leads to safeguard losses. As a player, I didn’t expect states and memorycard saves to be dependant (as a developer, yes). Because, as a player, save states are very convienent instant safeguards (as a developer, a full memory dump).
For example, if I decide to remove save states to make some space, it can completely erase my memorycard if I loaded a very old save state unintentionally before quitting the game. Yet, as a player again, I expect memorycard saves to be rock solid.

Is there something we can do to prevent any type of save loss in recalbox? Is this something you can even control? Should I discuss the matter with emulator developers?
